src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './components/UserInfo';
+import { Todo } from './components/TodoInfo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(false);
+  const [isUserValid, setIsUserValid] = useState(false);
+  const [user, setUser] = useState(0);
+  const [actualTodos, setTodos] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const validTitle = title.trim();
+
+    const newId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+    const addedTodo: Todo = {
+      id: newId,
+      completed: false,
+      user: getUserById(Number(user)),
+      title: validTitle,
+      userId: Number(user),
```

Comment:
Let's create new todo after all our checks. 
We don't need to create such a massive data object when there is a chance that we will not pass the test.